### ISSUE TITLE
report division by zero on const range

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust#52696]
I wonder if it makes sense to report a division by zero at compiletime if we divide by an element of a const range which we know covers zero:
````rust
	for i in -2..2 {
		let x = 1/i;
		println!("{}", x);
	}
````

**Comment by oli-obk** [comment]
That's significantly beyond the current capabilities of the const propagator.

I don't think this issue can be acted upon any time soon.

It would also require 

```rust
if x == 0 {
    println!("foo");
} else {
    println!("{}", 100 / x);
}
```

to know that the division cannot ever panic. Otherwise we'd get a lot of false positives.

I think such a feature should go through the RFC process to figure out a unified model to solve all these lints and optimizations.

**Comment by oli-obk** [comment]
That said. I wouldn't mind a clippy MIR lint that does a best-effort thing for this.

**Comment by oli-obk** [comment]
I don't think tracking this feature individually makes much sense. It will be something that will happen mostly automatically with improved const propagation.
